src/signature.rs

```diff
@@ -372,6 +389,21 @@ impl From<&[u8; SIGNATURE_SERIALIZED_LENGTH]> for Signature {
 
 }
 
+impl From<ed25519::Signature> for Signature {
+    fn from(sig: ed25519::Signature) -> Signature {
+        (&sig.to_bytes()).into()
+    }
+}
+
+impl TryFrom<Signature> for ed25519::Signature {
```

Comment:
Given our test suite, I think we can have the infallible conversion here. Or does ed25519's signature type have more stringent conditions? It would probably be a good idea to check against https://hdevalence.ca/blog/2020-10-04-its-25519am.

Reply:
I know too little about the cryptographic details to make a good judgement, but tracing from the code-path, ed25519-Signature does check whether "the s-Scalar fit[s] within the order of ~2^(252.5).".
I could not find such a check in this crates Signature/Scalar types, but please correct me if I'm wrong here.